### ISSUE TITLE
Doc improvement proposal - what happens after place an order.

### ISSUE THREAD

**Issue by bufke** [django-oscar/django-oscar-api#58]
Hello - When implementing the [Place an order](https://django-oscar-api.readthedocs.org/en/latest/usage/communicate_with_the_api.html#place-an-order-checkout) api I noticed a confirmation email is never sent. I presume this is because I am expected to create the payment logic myself and only send the email once payment is confirmed. 

For reference Oscar sends the email in a view after running place_order. See [here](https://github.com/django-oscar/django-oscar/blob/f5d4c174621371e1e4c173e8f850fea6c5d6e9be/src/oscar/apps/checkout/mixins.py#L116). The order api calls place_order but not the handle_successful_order method that involves sending the order.

My proposal is just to note this in the "place an order" section of the docs that it is expected the developer will call send_confirmation_message themselves once the order is final if they want the email sent. Unless there is a better way. I'm happy to send a pull request doc change if this my understanding is correct.

**Comment by maerteijn** [comment]
Hi Bufke, `handle_sucessful_order`  does a lot of things we don't want in the API, like flushing sessions and a http redirect:
```python
        self.send_confirmation_message(order, self.communication_type_code)

        # Flush all session data
        self.checkout_session.flush()

        # Save order id in session so thank-you page can load it
        self.request.session['checkout_order_id'] = order.id

        response = HttpResponseRedirect(self.get_success_url())
        self.send_signal(self.request, response, order)
        return response
```
However, the [`CheckoutSerializer`](https://github.com/django-oscar/django-oscar-api/blob/master/oscarapi/serializers/checkout.py#L187) has `OrderPlacementMixin` as a base class so we could call `send_confirmation_message` [here](https://github.com/django-oscar/django-oscar-api/blob/master/oscarapi/serializers/checkout.py#L272). If you don't want to send a confirmation message at this stage, then you just don't define any `order_placed` communication event type (or create your own `OrderPlacementMixin` class and override this method).

I think I'm going to add this :) Thanks for the ticket!

**Comment by maerteijn** [comment]
@bufke I decided not to call the ``handle_successfull_order`` method as it's too specific for the regular ``OrderPlacementMixin`` django view. See #62, so oscarapi will send a signal where you can send the confirmation email yourself (I updated the docs with this).

**Comment by maerteijn** [comment]
Released in 1.0.2
